### PR TITLE
DOC: Removed InvalidComparison and LossySetItemError from docs

### PR DESCRIPTION
**Pull request by DeaMariaLeon**
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

Towards https://github.com/pandas-dev/pandas/issues/37875

https://github.com/pandas-dev/pandas/pull/48088
https://github.com/pandas-dev/pandas/pull/48687

@jorisvandenbossche @mroeschke : Could you comment if this is OK please? thanks.
Suggested by @MarcoGorelli

### CODE REVIEW

**Comment by mroeschke** [comment]
I'm still slightly concerned that overtime that these exceptions could leak in public APIs. Maybe it's good to document that they should be internal errors?

**Comment by DeaMariaLeon** [comment]
Do you mean put them back to the documentation and write a note?

**Comment by mroeschke** [comment]
> Do you mean put them back to the documentation and write a note?

Yes, I would be partial to keep them in the docs and note in the docstrings that they are internal errors

**Comment by DeaMariaLeon** [comment]
@MarcoGorelli stylers? shouldn't we do it for errors?

**Comment by MarcoGorelli** [comment]
yeah I mean to add these two errors to that set

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: pandas/errors/__init__.py [hunk 1]
```diff
@@ -687,9 +687,9 @@ class LossySetitemError(Exception):
     """
     Raised when trying to do a __setitem__ on an np.ndarray that is not lossless.
 
-    Examples
+    Notes
     --------
```
Comment (MarcoGorelli) [inline pandas/errors/__init__.py:691]: this needs to be the same length as "Notes"

File: pandas/errors/__init__.py [hunk 2]
```diff
@@ -703,9 +703,9 @@ class InvalidComparison(Exception):
     """
     Exception is raised by _validate_comparison_value to indicate an invalid comparison.
 
-    Examples
+    Notes
     --------
```
Comment (MarcoGorelli) [inline pandas/errors/__init__.py]: same

File: pandas/errors/__init__.py (hunk 1 above)
Comment (DeaMariaLeon) [reply pandas/errors/__init__.py]: I saw it but I wasn't fast enough (before you saw it!), sorry.. doing silly errors like these.. 
I'm advancing at a painfully slow speed

Comment (MarcoGorelli) [reply pandas/errors/__init__.py]: 😄 no worries, I thought you'd probably seen it (or would have soon) but commented just in case

**Comment by DeaMariaLeon** [comment]
It looks green @MarcoGorelli

**Comment by MarcoGorelli** [comment]
> edit: I think I have to merge again

hey - was there a merge conflict? if not, no need to merge

**Review by MarcoGorelli — APPROVED** [review]
thanks @DeaMariaLeon

**Pull request merged**